client/state/ui/layout-focus/reducer.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * External dependencies
+ */
+import includes from 'lodash/includes';
+
+/**
+ * Internal dependencies
+ */
+import {
+	LAYOUT_SET_FOCUS,
+	LAYOUT_NEXT_FOCUS_ACTIVATE,
+	LAYOUT_SET_NEXT_FOCUS,
+	SERIALIZE,
+	DESERIALIZE,
+} from 'state/action-types';
+import layoutFocusModule from 'lib/layout-focus';
+
+const initialState = { current: null, previous: null, next: null };
+
+export default function layoutFocus( state = initialState, action ) {
+	const validAreas = [ 'content', 'sidebar', 'sites', 'preview' ];
+	const isValidArea = area => includes( validAreas, area );
+	switch ( action.type ) {
+		case LAYOUT_SET_FOCUS:
+			if ( ! isValidArea( action.area ) ) {
+				return state;
+			}
+			// Temporary while replacing module with this reducer
+			layoutFocusModule.set( action.area );
+			return Object.assign( {}, state, { current: action.area, previous: state.current } );
+		case LAYOUT_SET_NEXT_FOCUS:
+			if ( ! isValidArea( action.area ) ) {
+				return state;
+			}
+			// Temporary while replacing module with this reducer
+			layoutFocusModule.setNext( action.area );
+			return Object.assign( {}, state, { next: action.area } );
+		case LAYOUT_NEXT_FOCUS_ACTIVATE:
+			// If we don't have a change queued and the focus has changed
+			// previously, set it to `content`. This avoids having to set the
+			// focus to content on all navigation links because it becomes the
+			// default after focus has shifted.
+			if ( ! state.previous ) {
+				return state;
+			}
+			// Temporary while replacing module with this reducer
+			layoutFocusModule.next();
+			return Object.assign( {}, state, { current: state.next || 'content', previous: state.current, next: null } );
+		case SERIALIZE:
```

Comment:
`state.ui` serialization is [disabled for all subtrees automatically](https://github.com/Automattic/wp-calypso/blob/3e9e2b7c9208bb6d8e7b576972dfd345f6eab16a/client/state/ui/reducer.js#L118-L124).

Further, I find the [`createReducer` helper](https://github.com/Automattic/wp-calypso/blob/3e9e2b7c9208bb6d8e7b576972dfd345f6eab16a/client/state/utils.js#L27-L94) (docs lacking, cc @gziolo 😄 ) to be quite nice to avoid having to manually specify these cases.